### ISSUE TITLE
Invalid row number for failing tests

### ISSUE THREAD

**Issue by mschulzeLpz** [bitwes/Gut#130]
Hi at first it is very nice to have a unit testing support, thanks so lot.
After playing around with it i found some issue related to the row number for failing test.

here my simple test script
```
1. extends "res://addons/gut/test.gd"
2. 
3. func test_error_row():
4. 	assert_true( true, "is true")
5. 	assert_true( true, "is true")
6. 	assert_true( false, "is true")
```
The test run output is:
> /-----------------------------------------
>Running Script res://test/utils/test_error.gd
>-----------------------------------------/
>* test_error_row
>    FAILED:  is true
>      at line 3

The message says the error is located at line 3 which is pointing to the test func name and not to the real error line (6).
By bigger test functions it is hard to find the correct error without adding for each asssert an extra unique message.

**Comment by bitwes** [comment]
This is as designed.  GUT does very limited parsing of the test scripts.  It just finds all the test functions, it does not parse the functions themselves.  I added the line numbers to the output because I could, but I don't have any plans on making it smarter than it is.  Figuring out the actual line numbers would probably be very difficult.  I've contemplated removing the line numbers entirely since it doesn't show them for Inner Test Classes (they aren't parsed like other tests, they use Godot built-ins to find the function names).

When performing multiple asserts I consider it to be a best practice to add a message to each of the asserts so that the output is easier to read.

**Comment by mschulzeLpz** [comment]
Hi I implemented to grab the correct line number from the stack trace in the gut.gd script.
```
func _fail(text=''):
	_gui.add_failing()
	if(_current_test != null):
		var line_text = '  at line '
		var stackTrace = get_stack()
		if(stackTrace!=null):
			var line_number = _extractLineNumber(stackTrace, _current_test.name)
			line_text += str(line_number)
			p(line_text, LOG_LEVEL_FAIL_ONLY)
			# format for summary
			line_text =  "\n    " + line_text
		_new_summary.add_fail(_current_test.name, text + line_text)
		_current_test.passed = false

func _extractLineNumber(stackTrace:Array, testName) -> int:
	for index in stackTrace.size():
		var line:Dictionary = stackTrace[index]
		var function = line.get("function")
		if function == testName:
			return line.get("line")
	return -1
```

The output now is:

```
/-----------------------------------------
Running Script res://test/utils/test_errorLine.gd
-----------------------------------------/
* test_error_row
    FAILED:  is true
      at line 6
```

**Comment by bitwes** [comment]
That's great!  I didn't know we could get to the stacktrace.  Thanks.  If you want to make a pull request that would be great, otherwise I'll try to implement this in the near future.

**Comment by mschulzeLpz** [comment]
Ok nice i will do during this week.

**Comment by bitwes** [comment]
You need to fork the repo first to your own account.  You can then push to
your fork and open a pull request against my repo.

**Comment by mschulzeLpz** [comment]
ok got it ;)

**Comment by mschulzeLpz** [comment]
Hi @bitwes here the pull request https://github.com/bitwes/Gut/pull/131
I can't execute all test sucessfully, godot 3.2.0 is crashing during executing res://test/unit/test_doubler.gd.

Looks like GUT is not working on Godot 3.2.alpha

**Comment by bitwes** [comment]
Implemented in 6.8.1
